readme_docs/VOYAGE_STYLE_GUIDE.md

```diff
@@ -0,0 +1,2237 @@
+  
+  ```typescript
+  // bad
+  items.map(item => {
+    return Math.sqrt(item);
+  });
+  
+  // bad
+  items.map(item => Math.sqrt(item));
+  
+  // bad
+  getUsernames()
+    .then(usernames => {
+      processUsernames(usernames);
+    });
+  
+  // bad
+  getUsernames()
+    .then(usernames => processUsernames(usernames));
+  
+  // good
+  items.map(Math.sqrt);
+    
+  // good
+  getUsernames()
+    .then(processUsernames);
+  ```
+    
+## Iterators and Generators
+
+#### Don't use iterators
+
+  > Why? This enforces our immutable rule. Dealing with pure functions that return values is easier to reason about than side effects.
+
+  > Use `map()` / `every()` / `filter()` / `find()` / `findIndex()` / `reduce()` / `some()` / ... to iterate over arrays, and `Object.keys()` / `Object.values()` / `Object.entries()` to produce arrays so you can iterate over objects.
+
+```typescript
+const numbers = [1, 2, 3, 4, 5];
+
+// bad
+let sum = 0;
+for (let num of numbers) {
+  sum += num;
+}
+
+sum === 15;
+
+// good
+let sum = 0;
+numbers.forEach(num => sum += num);
+sum === 15;
+
+// best (use the functional force)
+const sum = numbers.reduce((total, num) => total + num, 0);
+sum === 15;
+```
+
+#### If you must use generators use proper spacing
+
+  > Why? `function` and `*` are part of the same conceptual keyword - `*` is not a modifier for `function`, `function*` is a unique construct, different from `function`.
+
+```typescript
+// bad
+function * foo() {
+}
+
+const bar = function * () {
+}
+
+const baz = function *() {
+}
+
+const quux = function*() {
+}
+
+function*foo() {
+}
+
+function *foo() {
+}
+
+// very bad
+function
+*
+foo() {
+}
+
+const wat = function
+*
+() {
+}
+
+// good
+function* foo() {
+}
+
+const foo = function* () {
+}
+```
+    
+**[⬆ back to top](#table-of-contents)**
+    
+## Properties
+
+#### Use dot notation
+
+```typescript
+const luke = {
+  jedi: true,
+  age: 28,
+};
+
+// bad
+const isJedi = luke['jedi'];
+
+// good
+const isJedi = luke.jedi;
+```
+
+#### Use brackets `[]` when accessing properties with a variable
+
+```typescript
+const luke = {
+  jedi: true,
+  age: 28,
+};
+
+function getProp(prop) {
+  return luke[prop];
+}
+
+const isJedi = getProp('jedi');
+```
+    
+**[⬆ back to top](#table-of-contents)**
+
+## Comparison Operators & Equality
+    
+#### Use `===` and `!==` over `==` and `!=`
+
+#### Truthiness
+
+  > Conditional statements such as the `if` statement evaluate their expression using coercion with the `ToBoolean` abstract method and always follow these simple rules:
+
+    + **Objects** evaluate to **true**
+    + **Undefined** evaluates to **false**
+    + **Null** evaluates to **false**
+    + **Booleans** evaluate to **the value of the boolean**
+    + **Numbers** evaluate to **false** if **+0, -0, or NaN**, otherwise **true**
+    + **Strings** evaluate to **false** if an empty string `''`, otherwise **true**
```

Comment:
This section is not rendering correctly. The indentation is making it into a code block. Did you mean to use a blockquote instead?